include/nbl/system/ISkippableAsyncQueueDispatcher.h

```diff
@@ -0,0 +1,66 @@
+#ifndef __NBL_I_SKIPPABLE_ASYNC_QUEUE_DISPATCHER_H_INCLUDED__
+#define __NBL_I_SKIPPABLE_ASYNC_QUEUE_DISPATCHER_H_INCLUDED__
+
+#include "nbl/system/IAsyncQueueDispatcher.h"
+#include "nbl/system/SReadWriteSpinLock.h"
+
+namespace nbl {
+namespace system
+{
+
+struct SSkippableRequestBase : impl::IAsyncQueueDispatcherBase::request_base_t
+{
+    void set_skip_no_lock()
+    {
+        skip = true;
+    }
+    void set_skip()
+    {
+        write_lock_guard<> lk(rwlock);
+        set_skip_no_lock();
+    }
```

Comment:
roll the `future` pointer and the value storage pointer into this class, then remove `set_skip_no_lock` method

and roll the request and future pointer nulling into the `set_skip` method (which should be named `abort` or `cancel`)

Reply:
resolved.